Source/Meadow.Foundation.Peripherals/Sensors.Temperature.Max31865/Driver/Max31865.cs

```diff
@@ -0,0 +1,425 @@
+        {
+            if (!IsSampling) { return; }
+            IsSampling = false;
+        }
+    }
+
+    public byte ReadFault()
+    {
+        return this.spiComms.ReadRegister(Registers.FaultStat);
+    }
+
+    private async Task<ushort> ReadInternal()
+    {
+        ClearFault();
+        Enable(Configuration.Bias);
+        await Task.Delay(10);
+
+        Enable(Configuration.OneShot);
+        await Task.Delay(65);
+
+        var rtd = this.spiComms.ReadRegisterAsUShort(Registers.RtdMsb, ByteOrder.LittleEndian);
+
+        // Disable bias current again to reduce self-heating.
+        Disable(Configuration.Bias);
+
+        // // remove the fault bit
+        rtd >>= 1;
+
+        return rtd;
+    }
+
+    private void ClearFault()
+    {
+        byte value = this.spiComms.ReadRegister(Registers.ConfigurationRead);
+
+        // Page 14 (Fault Status Clear (D1)) of the technical documentation
+        value = (byte)(value & ~0x2C);
+        value |= Configuration.FaultStat;
+
+        this.spiComms.WriteRegister(Registers.ConfigurationWrite, value);
+    }
+
+    private void Enable(byte configuration)
+    {
+        byte value = this.spiComms.ReadRegister(Registers.ConfigurationRead);
+
+        value |= configuration;
+
+        this.spiComms.WriteRegister(Registers.ConfigurationWrite, value);
+    }
+
+    private void Disable(byte configuration)
+    {
+        byte value = this.spiComms.ReadRegister(Registers.ConfigurationRead);
+
+        value &= (byte)~configuration;
+
+        this.spiComms.WriteRegister(Registers.ConfigurationWrite, value);
+    }
+
+    private void Initialize()
+    {
+        byte value = 
+            wires == Wires.ThreeWire ? Configuration.ThreeWire : Configuration.TwoFourWire |
+            conversionFilterMode == ConversionFilterMode.Filter50Hz ? Configuration.Filter50Hz : Configuration.Filter60Hz;
+
+        this.spiComms.WriteRegister(Registers.ConfigurationWrite, value);
+    }
+
+    /// <summary>
+    /// Releases unmanaged and - optionally - managed resources.
+    /// </summary>
+    /// <param name="disposing">
+    /// <c>true</c> to release both managed and unmanaged resources;
+    /// <c>false</c> to release only unmanaged resources.
+    /// </param>
+    protected virtual void Dispose(bool disposing)
+    {
+        if (!IsDisposed)
+        {
+            if (disposing)
+            {
+                if (portCreated)
+                {
+                    port?.Dispose();
+                }
+            }
+
+            IsDisposed = true;
+        }
+    }
+
+    /// <summary>
+    /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+    /// </summary>
+    public void Dispose()
+    {
+        // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+        Dispose(disposing: true);
+        GC.SuppressFinalize(this);
+    }
+}
+
+public interface IResistanceTemperatureDetectorConverter
```

Comment:
This might be a case of over engineering but if you look at the AN709_0.pdf data sheet that I have included there are multiple algorithms for calculating the temperature from the output of the RTD device. My initial thought was to use the Direct Mathematical Method which most other libraries support but leave it open to easy extension if you wanted. I would be happy to remove this concept and just use the conversion directly in the driver if you prefer?

Reply:
My gut would be to just have a virtual `Convert` on the class instead of the interface, so if someone wanted to change conversion math, they could subclass and override it.

Reply:
Good idea!